analysis_engine/td/fetch_api.py

```diff
@@ -30,49 +30,187 @@ def fetch_calls(
+        return ae_consts.EMPTY, pd.DataFrame([{}])
+
+    options_list = []
+
+    # assumes UTC conversion will work with the system clock
+    created_minute = (
+        datetime.datetime.utcnow() - datetime.timedelta(hours=5)).strftime(
+            '%Y-%m-%d %H:%M:00')
+    last_close_date = ae_utils.get_last_close_str(
+        fmt='%Y-%m-%d %H:%M:00')
 
+    # hit bug where dates were None
+    if not last_close_date:
+        last_close_date = created_minute
 
-def fetch(
-        fetch_type,
+    for node in org_records:
+        node['date'] = last_close_date
+        node['created'] = created_minute
+        node['ticker'] = ticker
+        if (
+                node['option_type'] == 'call' and
+                node['expiration_type'] == 'standard' and
+                float(node['bid']) > 0.01):
+            node['opt_type'] = int(ae_consts.OPTION_CALL)
+            node['exp_date'] = node['expiration_date']
+
+            new_node = {}
+            for col in td_consts.TD_OPTION_COLUMNS:
+                if col in node:
+                    if col in td_consts.TD_EPOCH_COLUMNS:
+                        # trade_date can be None
+                        if node[col] == 0:
+                            new_node[col] = None
+                        else:
+                            new_node[col] = ae_utils.epoch_to_dt(
+                                epoch=node[col]/1000,
+                                use_utc=False,
+                                convert_to_est=True).strftime(
+                                    ae_consts.COMMON_TICK_DATE_FORMAT)
+                            """
+                            Debug epoch ms converter:
+                            """
+                            """
+                            print('-----------')
+                            print(col)
+                            print(node[col])
+                            print(new_node[col])
+                            print('===========')
+                            """
+                        # if/else valid date
+                    else:
+                        new_node[col] = node[col]
+                    # if date column to convert
+                # if column is in the row
+            # convert all columns
+
+            options_list.append(new_node)
+    # end of records
+
+    full_df = pd.DataFrame(options_list).sort_values(
+        by=[
+            'strike'
+        ],
+        ascending=True)
+
+    num_chains = len(full_df.index)
+
+    df = None
+    if latest_close:
+        df_filter = (
+            (full_df['strike'] >=
+                (latest_close - 10.0)) &
+            (full_df['strike'] <=
+                (latest_close + 10.0)))
+        df = full_df[df_filter].copy().sort_values(
+            by=[
+                'date',
+                'strike'
+            ]).reset_index()
+    else:
+        mid_chain_idx = int(num_chains / 2)
+        low_idx = int(
+            mid_chain_idx - 200)
```

Comment:
do we want this 200 creating a 400 index spread to be adjustable and/or a const?

Reply:
done - i think some users may have the capacity to store all option chains and this is using an env var in consts.py